### PR TITLE
curve-fuse: return NOSPACE when volume space used up

### PR DESCRIPTION
**Pull request by wu-hanqing**
### What problem does this PR solve?

Issue Number: #1428

Problem Summary:

### What is changed and how it works?

What's Changed:

How it Works:

Side effects(Breaking backward compatibility? Performance regression?):

### Check List

- [ ] Relevant documentation/comments is changed or added
- [ ] I acknowledge that all my contributions will be made under the project's license

### CODE REVIEW

**Comment by wu-hanqing** [comment]
recheck

**Comment by wu-hanqing** [comment]
recheck

File: curvefs/src/volume/common.cpp [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+#include <ostream>
+#include <string>
+
+#include "curvefs/src/volume/utils.h"
+
+namespace curvefs {
+namespace volume {
+
+std::ostream& operator<<(std::ostream& os, const AllocateHint& hint) {
```
Comment (SeanHai) [inline curvefs/src/volume/common.cpp:33]: Implement this in AllocateHint maybe more clear.

Comment (wu-hanqing) [reply curvefs/src/volume/common.cpp:32]: it's not a core function, so I think it's ok to put it here. And, put so many logic in header file will slow down compilation and linking.

**Comment by wu-hanqing** [comment]
recheck

**Comment by wu-hanqing** [comment]
recheck

**Review by SeanHai — APPROVED** [review]

**Pull request merged**